web/docs/migrate-from-0-11-to-0-12.md

```diff
@@ -330,20 +332,206 @@ Once we confirm everything works well locally, we will apply the same changes to
+    import type { GetUserFieldsFn } from '@wasp/types'
+
+    // highlight-start
+    export const getUserFields: GetUserFieldsFn = async (_context, args) => {
+      const displayName = args.profile.displayName
+      return { displayName }
+    }
+    // highlight-end
+    ```
+
+    </TabItem>
+
+    <TabItem value="after" label="After">
+
+    ```wasp title="main.wasp"
+    app crudTesting {
+      // ...
+      auth: {
+        userEntity: User,
+        methods: {
+          google: {
+            // highlight-next-line
+            userSignupFields: import { fields } from "@src/server/auth/google.js",
+          },
+        },
+        onAuthFailedRedirectTo: "/login",
+      },
+    }
+    ```
+
+    ```ts title="src/server/auth/signup.ts"
+    // highlight-next-line
+    import { defineUserSignupFields } from 'wasp/server/auth'
+
+    // highlight-start
+    export const fields = defineUserSignupFields({
+      displayName: async (data) => {
+        if (!data.profile || !data.profile.displayName) {
+          throw new Error('Display name is not available')
+        }
+        return data.profile.displayName
+      },
+    })
+    // highlight-end
+    ```
+
+    Read more about the `userSignupFields` function [here](/auth/overview.md#1-defining-extra-fields).
+
+    </TabItem>
+    </Tabs>
+
+  </details>
+
 1. Ensure your **local development database is running**.
 1. **Do the schema migration** (create the new auth tables in the database) by running:
    ```bash
    wasp db migrate-dev
    ```
-You should see the new `Auth`, `AuthIdentity` and `Session` tables in your database. You can use the `wasp db studio` command to open the database in a GUI and verify the tables are there. At the moment, they will be empty (no data).
+You should see the new `Auth`, `AuthIdentity` and `Session` tables in your database. You can use the `wasp db studio` command to open the database in a GUI and verify the tables are there. At the moment, they will be empty.
 1. **Do the data migration** (move existing users from the old auth system to the new one by filling the new auth tables in the database with their data):
 
    1. **Implement your data migration function(s)** in e.g. `src/migrateToNewAuth.ts`.
 
       Below we prepared [examples of migration functions](#example-data-migration-functions) for each of the auth methods, for you to use as a starting point.
-      They should be fine to use as-is, meaning you can just copy them and they are likely to work out of the box for typical use cases, but you can also modify them to your needs (and should do so if needed).
+      They should be fine to use as-is, meaning you can just copy them and they are likely to work out of the box for typical use cases, but you can also modify them to your needs.
 
-      You will at the end want to have one function per each auth method that you use in your app.
+      You will want to have one function per each auth method that you use in your app.
```

Comment:
```suggestion
      You will want to end up with having one function per each auth method that you use in your app.
```

Reply:
This bit felt redundant, so I've removed it for better reading flow. It's probably subjective either way.

Reply:
yeah, the last one is better.
We are communicating here what is the expected end result, so I thought it is valuable to keep communicating this.

I feel like in situations like this, when it is not clear what is better, it doesn't hear to go with the option that provides a bit extra info to reader, even if a bit redundant, if that info is short and not in collision with the rest of the text. it won't hurt, and it can help.